### PR TITLE
(fix) O3-3897: Improvements to the 'all encounters' tab on the visits page

### PR DESCRIPTION
**Pull request by mccarthyaaron**
## Requirements

- [X] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-docs.openmrs.org/docs/frontend-modules/contributing.en-US#contributing-guidelines) label. See existing PR titles for inspiration.
- [ ] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://om.rs/o3ui).
- [ ] My work includes tests or is validated by existing tests.

## Summary
This PR aims at making the encounter tabs in the visists section of the patient chat implement the right scroll behaviour as per [this video](https://www.loom.com/share/7aa1bf26257e457f8ac3082c11d792a3?sid=fd1584da-9313-4644-861d-aaa6b71c5c68)

## Screenshots
https://github.com/user-attachments/assets/5e276a45-5bf0-4eb5-bbc4-28325d8ab959

## Related Issue
https://openmrs.atlassian.net/browse/O3-3897

### CODE REVIEW

**Comment by ciaranduffy** [comment]
@mccarthyaaron, thanks for the work on this. 

There are a few elements not quite right, visually though. 

Screenshot 2024-09-20 at 12 21 41

1. The tabs should be in a container that's 40px in height, with a 2px bottom border that's #E0E0E0 or `Border-subtle-00` in colour. 

2. The position of the tabs is not correct. They tabs should sit immediately below the vitals header in their container. The first visit card should be 16px from the bottom of the tabs container.

**Comment by mccarthyaaron** [comment]
@ciaranduffy I have made some changes. Though there is still an issue with "The tabs should sit immediately below the vitals header". That is being prevented by a margin applied generically to all slots as shown [here](https://github.com/openmrs/openmrs-esm-patient-chart/blob/main/packages/esm-patient-chart-app/src/patient-chart/chart-review/dashboard-view.scss#L14). Changing it would affect all other widgets(medications, orders, etc). We could do something like [this](https://github.com/openmrs/openmrs-esm-patient-chart/compare/main...mccarthyaaron:openmrs-esm-patient-chart:patch-1) , applying a different styles to the 'patient-chart-encounters-dashboard-slot'. Let me know what you think!

**Comment by ciaranduffy** [comment]
@mccarthyaaron, I'm not a developer so I don't really understand the code snippets you shared with me. Could you share a screenshot or a video of the end result on your side? Thanks

**Comment by ciaranduffy** [comment]
Thanks @mccarthyaaron, I think for now this is fine. If we need to add internal padding, when the visit summary cards themselves are iterated to match the design, hopefully that's a possibility? 

Thanks

**Comment by mccarthyaaron** [comment]
> Thanks @mccarthyaaron, I think for now this is fine. If we need to add internal padding, when the visit summary cards themselves are iterated to match the design, hopefully that's a possibility?
> 
> Thanks

Yes it is @ciaranduffy. So should I go ahead and commit the changes from the screen recording

**Comment by ciaranduffy** [comment]
From the design side, I have no objections. Thanks

**Comment by denniskigen** [comment]
Is there a zeplin screen I could reference for this work, @ciaranduffy?

File: packages/esm-patient-chart-app/src/patient-chart/chart-review/dashboard-view.scss [hunk 1]
```diff
@@ -14,6 +14,11 @@
   margin: 1.3125rem;
 }
 
+// See https://zpl.io/lrlmdq0 for the Visits dashboard design
+[data-extension-slot-name='patient-chart-encounters-dashboard-slot'] {
```
Comment (denniskigen) [inline packages/esm-patient-chart-app/src/patient-chart/chart-review/dashboard-view.scss:18]: https://github.com/openmrs/openmrs-esm-patient-chart/pull/2024#issuecomment-2365329831 addressed here.

**Review by denniskigen — APPROVED** [review]
LGTM. Thanks, @mccarthyaaron. The one thing that's outstanding here is blending the borders so that the green border of the active tab blends in with the 2px bottom border of the containing div.

**Pull request merged**